src/components/MessageCard/MessageCard.jsx

```diff
@@ -1,187 +1,100 @@
 import { noop } from '../../utilities/other'
-import Animate from '../Animate'
-import {
-  ActionUI,
-  BodyUI,
-  ImageContainerUI,
-  ImageUI,
-  MAX_IMAGE_SIZE,
-  MessageCardUI,
-  SubtitleUI,
-  TitleUI,
-} from './MessageCard.css'
-import Truncate from '../Truncate'
-
-const sizeWithRatio = (recalculatedSide, otherSide, defaultValue) =>
-  // Check if other side is smaller than max size to not recalculate unnecessarily this side as it doesn't need any scaling
-  // other condition checks that the image fits the boundaries
-  otherSide < MAX_IMAGE_SIZE
-    ? defaultValue
-    : (recalculatedSide / otherSide) * MAX_IMAGE_SIZE
-
-export class MessageCard extends React.PureComponent {
-  static className = 'c-MessageCard'
-  static Button = MessageCardButton
-
-  getClassName() {
-    const { align, className, isMobile, isWithBoxShadow } = this.props
-    return classNames(
-      MessageCard.className,
-      align && `is-align-${align}`,
-      className,
-      isMobile && 'is-mobile',
-      isWithBoxShadow && `is-with-box-shadow`
-    )
-  }
-
-  getTruncatedText(text, limit) {
-    return (
-      <Truncate limit={limit} type="end">
-        {text}
-      </Truncate>
-    )
-  }
-
-  renderTitle() {
-    const { title } = this.props
-    return title ? (
-      <TitleUI size="h4" data-cy="beacon-message-title">
-        {this.getTruncatedText(title, 110)}
-      </TitleUI>
-    ) : null
-  }
-
-  renderSubtitle() {
-    const { subtitle } = this.props
-    return subtitle ? (
-      <SubtitleUI
-        size="h5"
-        weight={500}
-        light
-        data-cy="beacon-message-subtitle"
-      >
-        {this.getTruncatedText(subtitle, 110)}
-      </SubtitleUI>
-    ) : null
-  }
-
-  renderBody() {
-    const { onBodyClick, title, subtitle } = this.props
-    let { body } = this.props
-    const withMargin = title || subtitle
-
-    // if there is no html in the string, transform new line to paragraph
-    if (body && !/<\/?[a-z][\s\S]*>/i.test(body)) {
-      body = body.split('\n').join('<br>')
-    }
-
-    return body ? (
-      <BodyUI
-        onClick={onBodyClick}
-        withMargin={withMargin}
-        data-cy="beacon-message-body-content"
-      >
-        <div dangerouslySetInnerHTML={{ __html: body }} />
-      </BodyUI>
-    ) : null
-  }
-
-  renderImage() {
-    const { image } = this.props
-
-    if (!image) {
-      return null
-    }
-
-    const { height, width } = this.calculateSize(image)
+import { MessageCardUI, MessageCardWrapperUI } from './MessageCard.css'
+import { MessageCardTitle } from './components/MessageCard.Title'
+import { MessageCardSubtitle } from './components/MessageCard.Subtitle'
+import { MessageCardImage } from './components/MessageCard.Image'
+import { MessageCardAction } from './components/MessageCard.Action'
+import { MessageCardBody } from './components/MessageCard.Body'
+import { MessageCardContent } from './components/MessageCard.Content'
+
+export const MessageCard = React.memo(
```

Comment:
I did take a look at the implementation in beacon and hs-app, `innerRef` is not used at all. That's a good occasion to migrate this component to forwardRef and remove the `innerRef` prop entirely.

After that I am good to go!

Reply:
Good catch, changing it now :)

Reply:
awesome thank you!!!